reddit_subreddit_block.py

```diff
@@ -1,16 +1,157 @@
+
+class SubredditSignal(Signal):
+    """ Translates attributes from subreddit posts into Signal format """
+    def __init__(self, data):
+        super().__init__()
+        for k in data:
+            setattr(self, k, data[k])
 
 @Discoverable(DiscoverableType.block)
-class Example(Block):
+class SubredditFeed(RESTPolling):
+
+    """ This block polls the Reddit feed for the most recent posts in subreddits that match the queries.
 
-    """ This is the Example block. Put a brief description here. """
+        Properties:
+            creds (APICredentials): API credentials
+    """
 
     version = VersionProperty('0.1.0')
 
-    def process_signals(self, signals, input_id='default'):
-        for signal in signals:
-            pass
-        self.notify_signals(signals, output_id='default')
+    URL_FORMAT = ("https://oauth.reddit.com/r/{}/new.json?before={}")
+
+    # the default number of posts returned by the Reddit API
+    DEFAULT_LIMIT = 25
+
+    creds = ObjectProperty(Creds, title='Credentials')
+
+    def __init__(self):
+        super().__init__()
+        self.post_ids = []
+
+    def configure(self, context):
+        """ Configures block, sets up a new instance for each query, and
+            captures the id of the most recent post.
+        """
+        super().configure(context)
+        for query in self.queries:
+            self.init_post_id(query);
+
+    def start(self):
+        super().start()
+
+    def _authenticate(self):
+        """ Overridden from RESTPolling block.
+
+            Sets up headers with OAuth token.
+        """
+        self.token = self.get_token();
+        headers = {"Authorization": "bearer " + self.token, "User-Agent": "nio"}
+
+        return headers
+
+    def init_post_id(self, query):
+        """ Makes the initial request to determine the id of the most recent
+            post, which becomes the 'before' query parameter, i.e., the marker
+            from where to start polling.
+
+            Args:
+                query (string): The name of the subreddit.
+
+        """
+        headers = self._authenticate()
+        response = requests.get(self.URL_FORMAT.format(query, None), headers=headers)
+        resp = response.json()
+        self.post_ids.append(resp['data']['children'][0]['data']['name'])
+
+    def get_token(self):
+        """ Gets token required for header in each request.
+
+            Returns:
+                access_token (string): An identifier that allows OAuth API access
+        """
+
+        client_auth = requests.auth.HTTPBasicAuth(self.creds.client_id, self.creds.app_secret)
+        post_data = {"grant_type": "password", "username": self.creds.app_username, "password": self.creds.app_password}
+        headers = {"User-Agent": "nio"}
+        response = requests.post("https://www.reddit.com/api/v1/access_token", auth=client_auth, data=post_data, headers=headers)
+
+        return response.json()['access_token']
+
+    def _get_post_id(self, signal):
```

Comment:
It makes sense to have a function to get the post ID from a signal, but we shouldn't override the rest polling block's function if we're changing the expected behavior. Here, we're getting the ID of a signal (not a post dictionary like the rest polling block is expecting).

So we should keep this function, but call it `_get_signal_id` or something like that to prevent the rest polling block from potentially breaking.